packages/compass-data-modeling/src/store/diagram.ts

```diff
@@ -732,6 +761,27 @@ export function addCollection(
   };
 }
 
+function renameFieldInRelationshipSide(
+  side: Relationship['relationship'][0],
+  edit: Extract<Edit, { type: 'RenameField' }>
+): Relationship['relationship'][0] {
+  if (
+    side.ns !== edit.ns ||
+    !side.fields ||
+    !isSameFieldOrAncestor(edit.field, side.fields)
+  ) {
+    return side;
+  }
+  return {
+    ...side,
+    fields: [
+      ...side.fields.slice(0, edit.field.length - 1),
+      edit.newName,
+      ...side.fields.slice(edit.field.length),
+    ],
+  };
+}
+
 function _applyEdit(edit: Edit, model?: StaticModel): StaticModel {
```

Comment:
This function and its helper methods start to get pretty big, we should move it out of this file so that it doesn't mix with reducer logic